### PR TITLE
don't trust rows of the "free" binary

### PR DESCRIPTION
**Pull request by untoreh**
- don't assume there is an actual block device
- provide a list of required packages

### CODE REVIEW

File: nench.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@
 # - runs IPv6 speedtest by default (if the server has IPv6 connectivity)
 # Run using `curl -s bench.wget.racing | bash`
 # or `wget -qO- bench.wget.racing | bash`
+# - list of possibly required packages: curl,gawk,coreutils,util-linux,procps,ioping
```
Comment (n-st) [inline nench.sh]: Nitpicky, but I'd move that to a separate commit. ;)

File: nench.sh [hunk 2]
```diff
@@ -185,12 +186,15 @@ uname -s -r -m
 printf '\n'
 
 printf 'Disks:\n'
-if command_exists lsblk
+if command_exists lsblk && [ -n "$(lsblk)" ]
```
Comment (n-st) [inline nench.sh]: Have you ever seen `lsblk` produce actually empty output?
According to its own manpage, it should always produce output (at least the header row), so this check wouldn't actually help here…

File: nench.sh [hunk 3]
```diff
@@ -185,12 +186,15 @@ uname -s -r -m
...
 then
     lsblk --nodeps --noheadings --output NAME,SIZE,ROTA --exclude 1,2,11 | sort | awk '{if ($3 == 0) {$3="SSD"} else {$3="HDD"}; printf("%-3s%8s%5s\n", $1, $2, $3)}'
 elif [ -r "/var/run/dmesg.boot" ]
 then
     awk '/(ad|ada|da|vtblk)[0-9]+: [0-9]+.B/ { print $1, $2/1024, "GiB" }' /var/run/dmesg.boot
+elif command_exists df
+then
+    df -h --output=source,fstype,size,itotal | awk 'NR == 1 || /^\/dev/'
```
Comment (n-st) [inline nench.sh]: This will include non-physical devices (`tmpfs`, NFS, Overlays, …).
I'd prefer to have a way to exclude those before merging this change.

Do you have a concrete case where the other two commands failed and the `df` solution was required?

File: nench.sh [hunk 4]
```diff
@@ -308,4 +312,4 @@ printf '%s\n' '-------------------------------------------------'
 
 # delete downloaded ioping binary if script has been run straight from a pipe
 # (rather than a downloaded file)
-[ -t 0 ] || rm -f ioping.static
+[ -t 0 ] || rm -f ioping.static
```
Comment (n-st) [inline nench.sh]: Missing newline at the end; might break the script on old plain-POSIX shells.

File: nench.sh (hunk 2 above)
Comment (untoreh) [reply nench.sh]: On ovz i don't see any output

File: nench.sh (hunk 3 above)
Comment (untoreh) [reply nench.sh]: ovz

File: nench.sh (hunk 2 above)
Comment (untoreh) [reply nench.sh]: it even gives exit code `1` if you prefer that

File: nench.sh (hunk 3 above)
Comment (n-st) [reply nench.sh]: Ah, I missed your `awk` filter. Should be fine then.
